http/util/src/main/java/org/wildfly/security/http/util/SetMechanismInformationMechanismFactory.java

```diff
@@ -110,7 +113,7 @@ public String getMechanismName() {
                         public String getHostName() {
                             return hostName;
                         }
-                    })});
+                    }), new RequestInformationCallback(request.getRequestURI())});
```

Comment:
Should we be handling the new `RequestInformationCallback` from this class? Wondering if it would make more sense to introduce a new `SetRequestionInformationMechanismFactory`. WildFly Core would need to be updated accordingly.

Reply:
@fjuma I was not sure at first since the `HttpServerAuthenticationMechanismFactory` is documented as `Factory to create authentication mechanisms` which did not seem related to logging or similar purpose, but it's true that this `requestURI` is a bit similar to already existing `SocketAddressCallbackServerMechanismFactory ` and its socket address handling. So I made the changes, let me know what you think? Thank you!

Reply:
@fjuma Here is affiliated commit for wildfly-core https://github.com/Skyllarr/wildfly-core/commit/f9cf8475072f84fae6ff0f4ef1de7f8172072db5 , if you like the changes I can create a WFCORE issue and submit this as a PR

Reply:
The Core changes look good to me.